core/amber/src/main/scala/edu/uci/ics/texera/web/resource/WorkflowCompilationResource.scala

```diff
@@ -2,60 +2,68 @@ package edu.uci.ics.texera.web.resource
 
 import com.typesafe.scalalogging.LazyLogging
 import edu.uci.ics.amber.engine.common.virtualidentity.WorkflowIdentity
-import edu.uci.ics.texera.Utils
 import edu.uci.ics.texera.web.model.websocket.request.LogicalPlanPojo
 import edu.uci.ics.texera.workflow.common.WorkflowContext
 import edu.uci.ics.texera.workflow.common.tuple.schema.Attribute
 import edu.uci.ics.texera.workflow.common.workflow.{PhysicalPlan, WorkflowCompiler}
 import org.jooq.types.UInteger
 
 import javax.annotation.security.RolesAllowed
-import javax.ws.rs.{Consumes, POST, Path, PathParam, Produces}
+import javax.ws.rs._
 import javax.ws.rs.core.MediaType
 
-case class WorkflowCompilationResponse(
-    physicalPlan: Option[PhysicalPlan],
-    operatorInputSchemas: Map[String, List[Option[List[Attribute]]]],
+trait WorkflowCompilationResponse
+case class WorkflowCompilationSuccess(
+    physicalPlan: PhysicalPlan,
+    operatorInputSchemas: Map[String, List[Option[List[Attribute]]]]
+) extends WorkflowCompilationResponse
+
+case class WorkflowCompilationFailure(
```

Comment:
Even if the schema propagation failed, the `operatorInputSchemas` should be there because the GUI still needs to show the propagated schema. So I think we should add the `operatorInputSchemas`

Reply:
aded `operatorInputSchemas` into `WorkflowCompilationFailure` as well.